### PR TITLE
Hosting request emails

### LINKED ISSUE #915: Email notifications for conversations in host requests
Seems the email notifications don't include messages in host requests.

### PR DESCRIPTION
**Pull request by aapeliv**
Implements emails for when hosting requests change state and when they get messages.

Closes #915, but more of a stopgap before we get a full push notification system built.

**Backend checklist**
- [x] Formatted my code by running `isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [x] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

File: app/backend/src/couchers/models.py [hunk 1]
```diff
@@ -117,6 +117,8 @@ class User(Base):
 
     # id of the last message that they received a notification about
     last_notified_message_id = Column(BigInteger, nullable=False, default=0)
+    # same as above for host requests
+    last_notified_request_message_id = Column(BigInteger, nullable=False, server_default=text("0"))
```
Comment (kalvdans) [inline app/backend/src/couchers/models.py:121]:
```suggestion
    last_notified_request_message_id = Column(BigInteger, nullable=False, server_default=0)
```

File: app/backend/src/couchers/urls.py [hunk 2]
```diff
@@ -30,10 +30,14 @@ def password_reset_link(password_reset_token):
...
 
-def host_request_link():
+def host_request_link_host():
     return f"{config['BASE_URL']}/messages/hosting/"
 
 
+def host_request_link_guest():
+    return f"{config['BASE_URL']}/messages/surfing/"
+
+
```
Comment (kalvdans) [inline app/backend/src/couchers/urls.py:40]: Update //docs/urls.md with these urls. Later on we want to have the conversation id in there too I guess.

File: app/backend/src/tests/test_references.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def create_host_request(
     )
     session.add(host_request)
     session.commit()
-    # send_host_request_email(host_request)
+    # send_new_host_request_email(host_request)
```
Comment (kalvdans) [inline app/backend/src/tests/test_references.py]:
```suggestion
```

**Review by kalvdans — COMMENTED** [review]
Some small comments, in general it looks good!

File: app/backend/src/couchers/models.py (hunk 1 above)
Comment (aapeliv) [reply app/backend/src/couchers/models.py:121]: SQLAlchemy won't take that... It needs to be an SQL text fragment! It's pretty dumb... I know...

Comment (kalvdans) [reply app/backend/src/couchers/models.py:121]: I saw 0 on the line above, but that's not a server_default. Are you saying we should change all default to server_default?

Comment (kalvdans) [reply app/backend/src/couchers/models.py:121]: LOL, "there's an [app](https://sqlalchemy-defaults.readthedocs.io/en/latest/) for that"

Comment (aapeliv) [reply app/backend/src/couchers/models.py:121]: > LOL, "there's an [app](https://sqlalchemy-defaults.readthedocs.io/en/latest/) for that"

🤦

Comment (aapeliv) [reply app/backend/src/couchers/models.py:121]: > I saw 0 on the line above, but that's not a server_default. Are you saying we should change all default to server_default?

Yeah, we probably should... Depending on it, `default` is python side, `server_default` is postgres side.

Probably most should be `server_default`s, but it's just, yeah, this text stuff is silly

**Review by kalvdans — APPROVED** [review]
There are no test for this, but all email formatting code is run by the unit tests I presume, and added to the outgoing queue. (Only the worker process is affected by the ENABLE_EMAIL config variable)

So the code should be free from syntax errors at least, LGTM!

**Pull request merged**